server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryResponse.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.io.stream;
+
+import org.opensearch.transport.TransportResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Extensibility support for Named Writeable Registry: response from extensions for name writeable registry entries
+ *
+ * @opensearch.internal
+ */
+public class NamedWriteableRegistryResponse extends TransportResponse {
+
+    private final Map<String, String> registry;
+
+    public NamedWriteableRegistryResponse(Map<String, String> registry) {
+        this.registry = registry;
```

Comment:
Caller still has access to `registry` and could alter the map.  Make a copy with `this.registry = new HashMap<>(registry);`